CriticalMass/LocationManager.swift

```diff
@@ -6,30 +6,57 @@
 //
 
 import CoreLocation
-import Foundation
 
-extension Location {
-    fileprivate init(_ clLocation: CLLocation, name: String? = nil, color: String? = nil) {
-        longitude = Float(clLocation.coordinate.longitude)
-        latitude = Float(clLocation.coordinate.latitude)
-        timestamp = Float(clLocation.timestamp.timeIntervalSince1970)
-        self.name = name
-        self.color = color
+class LocationManager: NSObject, CLLocationManagerDelegate, LocationProvider {
+    var accessPermission: LocationProviderPermission {
+        if Preferences.gpsEnabled {
+            switch CLLocationManager.authorizationStatus() {
+            case .authorizedAlways,
+                 .authorizedWhenInUse:
+                return .authorized
+            case .notDetermined:
+                return .unkown
+            case .restricted,
+                 .denied:
+                return .denied
```

Comment:
We don't need a default case because the switch statement is exhaustive. Thats one of the cool features of Swift that non exhaustive switch statements don't compile